[CI] RangeAggregatorTests#testSubAggCollectsFromManyBucketsIfManyRanges failure

**Build scan**:

https://gradle-enterprise.elastic.co/s/dunvomq3ngeca

**Repro line**:

```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.range.RangeAggregatorTests.testSubAggCollectsFromManyBucketsIfManyRanges" -Dtests.seed=25B8F134A6C94E53 -Dtests.security.manager=true -Dtests.locale=ar-YE -Dtests.timezone=Europe/Malta -Druntime.java=14 -Dtests.fips.enabled=true
```

**Reproduces locally?**:

yes

**Applicable branches**:

master & 7.x

**Failure history**:

It seems this is the first failure

**Failure excerpt**:

```
  2> java.lang.AssertionError: 
    Expected: <org.elasticsearch.search.aggregations.CardinalityUpperBound$KnownCardinalityUpperBound@391e356e>
         but: was <org.elasticsearch.search.aggregations.CardinalityUpperBound$KnownCardinalityUpperBound@7cd769bf>
        at __randomizedtesting.SeedInfo.seed([25B8F134A6C94E53:C1285C03439BF43C]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.search.aggregations.AggregatorTestCase$InternalAggCardinality.la
```

--- a/server/src/main/java/org/elasticsearch/search/aggregations/CardinalityUpperBound.java
+++ b/server/src/main/java/org/elasticsearch/search/aggregations/CardinalityUpperBound.java
@@ -33,7 +33,7 @@ public abstract class CardinalityUpperBound {
     /**
      * {@link Aggregator}s with this cardinality won't collect any data at
      * all. For the most part this happens when an aggregation is inside of a
-     * {@link BucketsAggregator} that is pointing to an unmapped field. 
+     * {@link BucketsAggregator} that is pointing to an unmapped field.
      */
     public static final CardinalityUpperBound NONE = new CardinalityUpperBound() {
         @Override
@@ -128,6 +128,19 @@ public abstract class CardinalityUpperBound {
         @Override
         public <R> R map(IntFunction<R> mapper) {
             return mapper.apply(estimate);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            KnownCardinalityUpperBound that = (KnownCardinalityUpperBound) o;
+            return estimate == that.estimate;
+        }
+
+        @Override
+        public int hashCode() {
+            return Integer.hashCode(estimate);
         }
     }
 }

